src/frontend/src/handler/alter_set_schema.rs

```diff
@@ -0,0 +1,206 @@
+            StatementType::ALTER_SINK => {
+                let (sink, old_schema_name) =
+                    catalog_reader.get_sink_by_name(db_name, schema_path, &real_obj_name)?;
+                session.check_privilege_for_drop_alter(old_schema_name, &**sink)?;
+                catalog_reader.check_relation_name_duplicated(
+                    db_name,
+                    &new_schema_name,
+                    &sink.name,
+                )?;
+                (Object::SinkId(sink.id.sink_id), old_schema_name)
+            }
+            StatementType::ALTER_CONNECTION => {
+                let (connection, old_schema_name) =
+                    catalog_reader.get_connection_by_name(db_name, schema_path, &real_obj_name)?;
+                session.check_privilege_for_drop_alter(old_schema_name, &**connection)?;
+                catalog_reader.check_connection_name_duplicated(
+                    db_name,
+                    &new_schema_name,
+                    &connection.name,
+                )?;
+                (Object::ConnectionId(connection.id), old_schema_name)
+            }
+            StatementType::ALTER_FUNCTION => {
+                let (function, old_schema_name) = if let Some(args) = func_args {
+                    let mut arg_types = Vec::with_capacity(args.len());
+                    for arg in args {
+                        arg_types.push(bind_data_type(&arg.data_type)?);
+                    }
+                    catalog_reader.get_function_by_name_args(
+                        db_name,
+                        schema_path,
+                        &real_obj_name,
+                        &arg_types,
+                    )?
+                } else {
+                    let (functions, old_schema_name) = catalog_reader.get_functions_by_name(
+                        db_name,
+                        schema_path,
+                        &real_obj_name,
+                    )?;
+                    if functions.len() > 1 {
+                        return Err(ErrorCode::CatalogError(format!("function name {real_obj_name:?} is not unique\nHINT: Specify the argument list to select the function unambiguously.").into()).into());
+                    }
+                    (
+                        functions.into_iter().next().expect("no functions"),
+                        old_schema_name,
+                    )
+                };
+                session.check_privilege_for_drop_alter(old_schema_name, &**function)?;
+                catalog_reader.check_function_name_duplicated(
+                    db_name,
+                    &new_schema_name,
+                    &function.name,
+                    &function.arg_types,
+                )?;
+                (
+                    Object::FunctionId(function.id.function_id()),
+                    old_schema_name,
+                )
+            }
+            _ => unreachable!(),
+        }
+    };
+
+    if old_schema_name == new_schema_name {
```

Comment:
If the schema name is unchanged, I guess it will return an error in `check_relation_name_duplicated` before. If so the check should be done in advance.